.github/workflows/backend_tests.yml

```diff
@@ -33,6 +35,121 @@ jobs:
       - name: Install coverage and configparser
         if: startsWith(github.head_ref, 'update-changelog-for-release') == false
         run: pip install coverage configparser
-      - name: run backend
+      - name: Run backend test shard
         if: startsWith(github.head_ref, 'update-changelog-for-release') == false
-        run: PYTHONIOENCODING=utf-8 python -m scripts.run_backend_tests --generate_coverage_report --exclude_load_tests
+        run: PYTHONIOENCODING=utf-8 python -m scripts.run_backend_tests --generate_coverage_report --ignore_coverage --exclude_load_tests --test_shard ${{ matrix.shard }}
+      - name: Upload coverage report
+        if: startsWith(github.head_ref, 'update-changelog-for-release') == false
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ format('backend_test_coverage_shard_{0}', matrix.shard) }}
+          path: .coverage
+  check_combined_coverage:
```

Comment:
check_combined_coverage check looks too big (and lots of duplicate patterns/lines), was there any issue with having a script that will do this? (not sure!)

Reply:
There were issues with using a for loop here, but github actions was also having issues at the time. I can try that again

Reply:
Actually no, this part cannot be simplified. I couldn't find a way to have a for loop in the github actions yaml file. Note that all the download steps (most of the duplicated code) are in github actions' workflow syntax, which doesn't support for loops as far as I could tell.

In 88d235465 I start using a for loop for the bash code, which reduces the duplication a little